source/extensions/network/dns_resolver/apple/apple_dns_impl.cc

```diff
@@ -378,6 +378,8 @@ void AppleDnsResolverImpl::PendingResolution::onDNSServiceGetAddrInfoReply(
   if (!(flags & kDNSServiceFlagsMoreComing) && isAddressFamilyProcessed(kDNSServiceProtocol_IPv4) &&
       isAddressFamilyProcessed(kDNSServiceProtocol_IPv6)) {
     ENVOY_LOG(debug, "DNS Resolver flushing queries pending callback");
+    pending_response_.status_ = ResolutionStatus::Completed;
+    pending_response_.details_ = "apple_dns_success";
```

Comment:
We discovered some instances of `not_set` in our testing, which gives the impression that the DNS query never gets executed. This is because by default, the status is set to `Completed` and details to `not_set`. This PR aims to fix the potential confusion.

I also updated the PR to add missing tests for the DNS resolution details that previously didn't exist.